ddl/backfilling_operator.go

```diff
@@ -0,0 +1,522 @@
+			writer, err := engine.CreateWriter(writerID, index.Meta().Unique)
+			if err != nil {
+				logutil.Logger(ctx).Error("create index ingest worker failed", zap.Error(err))
+				return nil
+			}
+			return &indexIngestWorker{
+				ctx:          ctx,
+				tbl:          tbl,
+				index:        index,
+				copCtx:       copCtx,
+				se:           nil,
+				sessPool:     sessPool,
+				writer:       writer,
+				srcChunkPool: srcChunkPool,
+			}
+		})
+	return &IndexIngestOperator{
+		AsyncOperator: operator.NewAsyncOperator[IndexRecordChunk, IndexWriteResult](ctx, pool),
+	}
+}
+
+type indexIngestWorker struct {
+	ctx context.Context
+
+	tbl   table.PhysicalTable
+	index table.Index
+
+	copCtx   *copContext
+	sessPool opSessPool
+	se       *session.Session
+
+	writer       ingest.Writer
+	srcChunkPool chan *chunk.Chunk
+}
+
+func (w *indexIngestWorker) HandleTask(rs IndexRecordChunk, sender func(IndexWriteResult)) {
+	result := IndexWriteResult{
+		ID:  rs.ID,
+		Err: rs.Err,
+	}
+	if w.se == nil {
+		sessCtx, err := w.sessPool.Get()
+		if err != nil {
+			result.Err = err
+			sender(result)
+			return
+		}
+		w.se = session.NewSession(sessCtx)
+	}
+	defer func() {
+		w.srcChunkPool <- rs.Chunk
+	}()
+	if result.Err != nil {
+		logutil.Logger(w.ctx).Error("encounter error when handle index chunk",
+			zap.Int("id", rs.ID), zap.Error(rs.Err))
+		sender(result)
+		return
+	}
+	count, nextKey, err := w.WriteLocal(&rs)
+	if err != nil {
+		result.Err = err
+		sender(result)
+		return
+	}
+	if count == 0 {
+		logutil.Logger(w.ctx).Info("finish a cop-request task", zap.Int("id", rs.ID))
+		sender(result)
+		return
+	}
+	result.Added = count
+	result.Scanned = count
+	result.Next = nextKey
+	if ResultCounterForTest != nil && result.Err == nil {
+		ResultCounterForTest.Add(1)
+	}
+	sender(result)
+}
+
+func (*indexIngestWorker) Close() {
+}
+
+// WriteLocal will write index records to lightning engine.
+func (w *indexIngestWorker) WriteLocal(rs *IndexRecordChunk) (count int, nextKey kv.Key, err error) {
+	oprStartTime := time.Now()
+	vars := w.se.GetSessionVars()
+	cnt, lastHandle, err := writeChunkToLocal(w.writer, w.index, w.copCtx, vars, rs.Chunk)
+	if err != nil || cnt == 0 {
+		return 0, nil, err
+	}
+	logSlowOperations(time.Since(oprStartTime), "writeChunkToLocal", 3000)
+	nextKey = tablecodec.EncodeRecordKey(w.tbl.RecordPrefix(), lastHandle)
+	return cnt, nextKey, nil
+}
+
+type indexWriteResultSink struct {
+	ctx context.Context
+
+	errGroup errgroup.Group
+	source   operator.DataChannel[IndexWriteResult]
+}
+
+func newIndexWriteResultSink(
+	ctx context.Context,
+) *indexWriteResultSink {
+	return &indexWriteResultSink{
+		ctx:      ctx,
+		errGroup: errgroup.Group{},
+	}
+}
+
+func (s *indexWriteResultSink) SetSource(source operator.DataChannel[IndexWriteResult]) {
+	s.source = source
+}
+
+func (s *indexWriteResultSink) Open() error {
+	s.errGroup.Go(s.collectResult)
+	return nil
+}
+
+func (s *indexWriteResultSink) collectResult() error {
```

Comment:
Will we enhance it to control the worker num of previous operator?
How do we deal with the result?

Reply:
I will the update reorg info with result in next PR.